openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -652,6 +652,21 @@ def _apply_cdn_to_url(self, url, base_url):
 
         return urlunparse((None, base_url, path, params, query, fragment))
 
+    def get_course_mode(self, mode_slug):
+        """
+        Find the specified mode associated with this course.
+
+        Returns:
+            The course mode if available, or
+        """
+        # In some code paths, the modes have been prefetched. Instead of calling get() directly here, which will
+        # miss the cache, instead just iterate through the list of prefetched modes and find the verified mode.
+        for mode in self.modes.all():
```

Comment:
What about the code path where the modes aren't prefetched?  Should this method be called only when it's prefetched?

Reply:
If so, can we rename it to be clear that it only gets the value if it's already prefetched?  Maybe: `get_prefetched_course_mode`.

Reply:
Done